mtp_api/apps/mtp_auth/serializers.py

```diff
@@ -177,11 +196,13 @@ def update(self, user, validated_data):
         updating_user = self.context['request'].user
         was_user_admin = user.groups.filter(pk=user_admin_group.pk).exists()
 
-        if user.pk == updating_user.pk and ('user_admin' in validated_data or 'role' in validated_data):
+        disallowed_self_updates = ('user_admin', 'role', 'is_locked_out')
```

Comment:
Possibly this should be reversed to `allowed_self_updates` - I think the list of things people should be able to update themselves is shorter than what they shouldn't.